packages/stream_chat/lib/src/core/api/sort_order.dart

```diff
@@ -104,13 +106,15 @@ class SortOption<T extends ComparableFieldProvider> {
   /// - 1 if the first object is greater than the second
   /// - -1 if the first object is less than the second
   ///
-  /// Handles null values by treating null as less than any non-null value.
+  /// Handles null values according to the nullOrdering setting.
+  /// NULLS FIRST puts nulls at the beginning regardless of sort direction.
+  /// NULLS LAST puts nulls at the end regardless of sort direction.
   ///
   /// ```dart
   /// final sortOption = SortOption<ChannelState>("last_message_at");
   /// final sortedChannels = channels.sort(sortOption.compare);
   /// ```
-  int compare(T a, T b) => direction * comparator(a, b);
+  int compare(T a, T b) => comparator(a, b);
 
   /// Returns a comparator function for sorting objects of type T.
   @JsonKey(includeToJson: false, includeFromJson: false)
```

Comment:
This changes breaks it if you have your own comparator (line 118/122)

```dart
if (_comparator case final comparator?) return comparator;
```

Because now direction has no effect anymore and you have to apply this yourself.

Maybe we could change that into something like this:
```dart
if (_comparator case final comparator?) return (a, b) => direction*comparator(a,b);
```

That still wouldn't apply `nullOrdering`, but at least it wouldn't break current behaviour.

Reply:
Updated